Add message to Swagger UI page encouraging visitors to use Runtime API instead

eecavanna opened the pull request:
On this branch, I made the following changes:
1. Converted the HTML description string into a Markdown description string. The docstring of the `FastAPI` constructor says that its `description` parameter can be a Markdown string (and I did not see anything about it being able to be an HTML string; indeed, I confirmed arbitrary HTML code—such as `<style>` tags—within the description did not seem to take effect). Note that the previous string, even with its `<a>` tag, may have been "sufficiently Markdown-y" already.
2. Refactored the description code so that the description is loaded from a template file, rather than being written in `app.py`. The template file uses an arbitrary `{{ name }}` syntax for placeholders, and `app.py` takes care of replacing those placeholders with their final values. With the current description, the values could have been hard-coded in the description, but I could see this being useful for other descriptions in the future (in case reviewers don't feel comfortable with this additional complexity, I'd be OK removing it and hard-coding the URLs in the description already). The placeholder syntax (and the helper function used to load the template) were copied from the `refscan` repo.
3. Added a message to the description, saying that this URL was designed to be accessed only by specific NMDC programs, and guiding visitors to use the Runtime API instead. This was my primary goal with this PR.

eecavanna commented:
**Note to Reviewers:** The content of the message will change (based upon feedback I received during today's infrastructure meeting, and—I expect—some follow-on discussions on Slack). I would still like for `nmdc-server` maintainers to review the code changes. I will unassign the Runtime squad members as reviewers, since the content is still under development.

eecavanna commented on nmdc_server/swagger_ui/templates/description.template.md:
```diff
@@ -0,0 +1,18 @@
+
+This API was designed to be accessed by
+the [NMDC Data Portal]({{ nmdc_data_portal_url }}) and
+the [NMDC Submission Portal]({{ nmdc_submission_portal_url }}) only.
```
During today's infrastructure meeting, @naglepuff suggested omitting the word "only" from this statement.

eecavanna commented:
A team member in attendance at today's infrastructure meeting expressed concern that the messaging we use might (incorrectly) imply that anyone can access anyone else's submission portal submissions.

naglepuff commented on nmdc_server/swagger_ui/templates/description.template.md:
```diff
@@ -0,0 +1,18 @@
+
+This API was designed to be accessed by
+the [NMDC Data Portal]({{ nmdc_data_portal_url }}) and
+the [NMDC Submission Portal]({{ nmdc_submission_portal_url }}) only.
```
I still have the same thoughts that I did on Thursday. Yes it is true that the API for `nmdc-server` is primarily designed with the data/submission portal in mind. However, I don't want to _discourage_ people from using the endpoints in other ways. We could even point them towards [this wiki](https://github.com/microbiomedata/nmdc-server/wiki/Search-API-Docs), which contains documentation about the search API (although this might be outdated. I haven't read through it in a while.

eecavanna replied:
Thanks, @naglepuff.

> However, I don't want to discourage people from using the endpoints in other ways.

I think some team members _do_ want to discourage people from accessing these endpoints/this API. I assume that is so team members don't feel obligated to (a) provide support to those people about their use of these endpoints, or (b) hold off on making breaking changes that would better serve the Data/Submission Portal because someone might depend upon the current behavior.

eecavanna commented on nmdc_server/app.py:
```diff
@@ -41,14 +42,20 @@ def create_app(env: typing.Mapping[str, str]) -> FastAPI:
+        .replace("{{ runtime_api_url }}", "https://api.microbiomedata.org")
+        .replace("{{ nmdc_data_portal_url }}", "https://data.microbiomedata.org")
+        .replace(
+            "{{ nmdc_submission_portal_url }}", "https://data.microbiomedata.org/submission/home"
```
Done via commit https://github.com/microbiomedata/nmdc-server/pull/1715/commits/67a86775dcba96fb0f5a5098237037bfedb55574.

eecavanna commented:
I've updated the message based upon all the feedback I've received so far. I updated the screenshot in the PR description above to show the latest message.

I plan to merge this PR in early next week and then solicit follow-up feedback from team members. I think seeing it "live" on the web (on the dev site), in context, will help some of them internalize it (compared to seeing the static screenshot above).
